BDArmory/Control/BDModuleOrbitalAI.cs

```diff
@@ -723,27 +751,33 @@ void Maneuver()
                     }
                     break;
             }
+            AlignEnginesWithThrust();
             UpdateRCSVector(rcsVector);
             UpdateBurnAlignmentTolerance();
         }
 
         void KillVelocity(bool onIntercept = false)
         {
+            // Slow down to KillVelocityTargetSpeed(). If on an intercept to within gun range, try to slow down as late as possible
+            // Otherwise, keep burning until at target speed
+
+            Vector3 relPos = targetVessel.CoM - vessel.CoM;
             Vector3 relVel = targetVessel.GetObtVelocity() - vessel.GetObtVelocity();
             float targetSpeed = KillVelocityTargetSpeed();
             bool maintainThrottle = (relVel.sqrMagnitude > targetSpeed * targetSpeed);
+            fc.thrustDirection = (relVel + targetVessel.perturbation).normalized;
+            bool useReverseThrust = !UseForwardThrust(fc.thrustDirection) && (Vector3.Dot(relPos, relVel) < 0f);
             if (onIntercept)
             {
-                Vector3 relPos = targetVessel.CoM - vessel.CoM;
                 Vector3 relAccel = targetVessel.perturbation - vessel.perturbation;
                 Vector3 toIntercept = Intercept(relPos, relVel);
                 float distanceToIntercept = toIntercept.magnitude;
                 float timeToIntercept = vessel.TimeToCPA(toIntercept, targetVessel.Velocity(), targetVessel.perturbation);
                 float cpaDistSqr = AIUtils.PredictPosition(relPos, relVel, relAccel, timeToIntercept).sqrMagnitude;
-
+                float interceptRangeMargin = Mathf.Min(interceptRanges.z * (1f + interceptMargin + (useReverseThrust ? 0.25f : 0f)), interceptRanges.y); // Extra margin when reverse thrusting since still facing target
                 if (cpaDistSqr < weaponManager.gunRange * weaponManager.gunRange) // Gun range intercept, balance between throttle actions and intercept accuracy
-                    maintainThrottle = relPos.sqrMagnitude < ((1f + interceptMargin) * (1f + interceptMargin) * interceptRanges.z * interceptRanges.z) || // Within intercept range margin
-                        (timeToIntercept == 0f) || // Moving away from target
+                    maintainThrottle = relPos.sqrMagnitude < (interceptRangeMargin * interceptRangeMargin) || // Within intercept range margin
+                        (maintainThrottle && Vector3.Dot(relPos, relVel) > 0f) || // Moving away from target faster than target speed, timeToIntecept == 0 does not always work here because it's possible we are on an orbit with two intersection points
```

Comment:
Keep in mind that `vessel.TimeToCPA` isn't going to correspond too accurately with actual orbits over long times due to the changing relative acceleration (i.e., the orbits are approximated as parabolas with constant acceleration instead of ellipses with inward pointing acceleration). Though it should stay reasonable for a few degrees (100km or so around Kerbin).

You might want to set an upper limit to `maxTime` to avoid false positives for `cpaDistSqr < weaponManager.gunRange * weaponManager.gunRange` due to secondary intercepts. Something like `vessel.orbit.period / 4` (or maybe less?) should be plenty to avoid picking up intercepts on the far side of the planet.